docs/sql-ref-literals.md

```diff
@@ -21,14 +21,74 @@ license: |
 
 A literal (also known as a constant) represents a fixed data value. Spark SQL supports the following literals:
 
+ * [Typed Literal for Binary, Date, Timestamp, and Interval](#typed-literal)
  * [String Literal](#string-literal)
  * [Binary Literal](#binary-literal)
  * [Null Literal](#null-literal)
  * [Boolean Literal](#boolean-literal)
  * [Numeric Literal](#numeric-literal)
+ * [Timestamp Literal](#timestamp-literal)
  * [Datetime Literal](#datetime-literal)
  * [Interval Literal](#interval-literal)
 
+### Typed Literal for Binary, Date, Timestamp, and Interval
+
+A typed Literal expression parsing string values to typed Literal 
+
+#### Syntax
+
+```sql
+[TYPE] '[VALUE]'
+```
+
+#### Parameters
+
+* **TYPE**
+   
+    `TYPE` parameter shows which type of Literal we will generate. Currently Date, Timestamp, Interval and Binary typed literal are supported.
+     
+      X: Binary Literal
```

Comment:
Ah, wait.  The `Binary Literal` section already has examples for this case though, why do we need this new section? In my comment https://github.com/apache/spark/pull/30421#discussion_r530024114, I just asked if `a type constructed value` is a common word in the doc.

Reply:
> Ah, wait. The `Binary Literal` section already has examples for this case though, why do we need this new section? In my comment [#30421 (comment)](https://github.com/apache/spark/pull/30421#discussion_r530024114), I just asked if `a type constructed value` is a common word in the doc.

That is what I mentioned in  https://github.com/apache/spark/pull/30496#issuecomment-733536446.

We need to introduce typed literal , this special  grammar. right?